source/common/http/http2/codec_impl_legacy.cc

```diff
@@ -269,16 +269,25 @@ void ConnectionImpl::StreamImpl::pendingRecvBufferLowWatermark() {
 
 void ConnectionImpl::ClientStreamImpl::decodeHeaders(bool allow_waiting_for_informational_headers) {
   auto& headers = absl::get<ResponseHeaderMapPtr>(headers_or_trailers_);
-  if (allow_waiting_for_informational_headers &&
-      CodeUtility::is1xx(Http::Utility::getResponseStatus(*headers))) {
-    waiting_for_non_informational_headers_ = true;
+
+  bool is_100_continue_header = false;
+  if (allow_waiting_for_informational_headers) {
+    const uint64_t status = Http::Utility::getResponseStatus(*headers);
+    if (CodeUtility::is1xx(status)) {
+      waiting_for_non_informational_headers_ = true;
```

Comment:
IMO it's confusing that waiting_for_non_informational_headers is false when a stream is created, when we are actually waiting for headers at that point.  Given we've changed the use here, what do you think about just making it 
received_noninformational_headers_ , initializing to false, setting to true when getting !1xx, and sending trailers iff(received_noninformational)?